Cortex/cortex_module/cortex_edr_connector.py

```diff
@@ -0,0 +1,170 @@
+
+    @cached_property
+    def pagination_limit(self) -> int:
+        return max(self.configuration.chunk_size, 100)
+
+    @cached_property
+    def client(self) -> ApiClient:
+        return ApiClient(
+            self.module.configuration.api_key,
+            self.module.configuration.api_key_id,
+        )
+
+    def split_alerts_events(self, alerts: List[Any]) -> List[str]:
+        """Split events from alerts and put them in the same list"""
+
+        combined_data = []
+        for alert in alerts:
+            shared_id = alert.get("alert_id")
+            events = alert.get("events")
+            del alert["events"]
+            combined_data.append(orjson.dumps(alert).decode("utf-8"))
+            for event in events:
+                event["external_id"] = shared_id
+                combined_data.append(orjson.dumps(event).decode("utf-8"))
+
+        return combined_data
+
+    def get_alerts_events_by_offset(self, offset: int, creation_time: int, pagination: int) -> Tuple[int, List[Any]]:
+        """Requests the Cortex API using the offset"""
+
+        search_from, serch_to = offset, offset + pagination
+        self.query["request_data"]["search_from"] = search_from
+        self.query["request_data"]["search_to"] = serch_to
+        self.query["request_data"]["filters"][0]["value"] = creation_time
+
+        response_query = self.client.post(url=self.alert_url, json=self.query).json().get("reply")
+        combined_data = self.split_alerts_events(response_query.get("alerts"))
+
+        return response_query.get("total_count"), combined_data
+
+    def get_all_alerts(self, pagination: int) -> None:
+        """Get all Cortex alerts from the API"""
+
+        total_alerts, combined_data = self.get_alerts_events_by_offset(0, self.timestamp_cursor, pagination)
+
+        if total_alerts > 0:
+            INCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(total_alerts)
+            self.log(message=f"Sending batch number 1 of {len(combined_data)} data", level="info")
+            OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(combined_data))
+            self.push_events_to_intakes(events=combined_data)
+
+            if total_alerts > pagination:
+                count_batch = 2
+                offset = pagination
+                while total_alerts > offset:
+                    total_alerts, new_combined_data = self.get_alerts_events_by_offset(
+                        offset, self.timestamp_cursor, pagination
+                    )
+                    OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(new_combined_data))
+                    self.log(
+                        message=f"Sending batch number {count_batch} of {len(new_combined_data)} events", level="info"
+                    )
+                    combined_data.extend(new_combined_data)
+                    offset += pagination
+                    count_batch += 1
```

Comment:
I think the complexity of this code can be decreased:
```python
   has_more_events = True
   page_number = 0

   while has_more_events:
      total_alerts, combined_data = self.get_alerts_events_by_offset(page_number*pagination, self.timestamp_cursor, pagination)
      has_more_events = total_alerts > (page_number + 1) * pagination
      page_number += 1
      
      OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(combined_data))
      self.push_events_to_intakes(events=combined_data)
```

We can also improve the performance by accumulating the events:
```python
has_more_events = True
page_number = 0
events= []

while has_more_events:
  total_alerts, combined_data = self.get_alerts_events_by_offset(page_number*pagination, self.timestamp_cursor, pagination)
  has_more_events = total_alerts > (page_number + 1) * pagination
  page_number += 1
  events.extend(combined_data)

  if len(events) > THRESHOLD:
      OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(events))
      self.push_events_to_intakes(events=events)

if len(events) > 0:
  OUTCOMING_EVENTS.labels(intake_key=self.configuration.intake_key).inc(len(events))
  self.push_events_to_intakes(events=events)
```